scripts/core/editor3/components/Editor3.jsx

```diff
@@ -0,0 +1,224 @@
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import {Editor, EditorState, RichUtils, ContentState} from 'draft-js';
+
+export class Editor3 extends React.Component {
+    constructor(props) {
+        super(props);
+        var initialContentState = ContentState.createFromText(props.value);
+        this.state = {editorState: EditorState.createWithContent(initialContentState)};
+
+        this.readOnly = props.readOnly || false;
+        this.showToolbar = props.showToolbar || false;
+        this.editorFormat = props.editorFormat || [];
+        this.parentOnChange = props.onChange;
+
+        this.focus = () => this.refs.editor.focus();
+        this.onChange = (editorState) => this._onChange(editorState);
+
+        this.handleKeyCommand = (command) => this._handleKeyCommand(command);
+        this.onTab = (e) => this._onTab(e);
+        this.toggleBlockType = (type) => this._toggleBlockType(type);
+        this.toggleInlineStyle = (style) => this._toggleInlineStyle(style);
```

Comment:
The arrow function binds all those functions to "this"
An alternative is to do:
this.toggleBlockType = this._toggleBlockType.bind(this);
or
this.toggleBlockType = this.toggleBlockType.bind(this);
With this pattern it looks a little nicer, depends by test of course.